tsdb/db.go

```diff
@@ -816,7 +825,10 @@ func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
-	db.compactor, err = NewLeveledCompactorWithChunkSize(ctx, r, l, rngs, db.chunkPool, opts.MaxBlockChunkSegmentSize, nil)
+	db.compactor, err = NewLeveledCompactorWithOptions(ctx, r, l, rngs, db.chunkPool, LeveledCompactorOptions{
+		MaxBlockChunkSegmentSize:    opts.MaxBlockChunkSegmentSize,
+		EnableOverlappingCompaction: opts.EnableOverlappingCompaction,
```

Comment:
This is making it optional for prometheus but for prometheus we want this blocks to be compacted always. Its only optional for other users of the TSDB like Thanos Cortex or Mimir.

Reply:
How would you propose to solve this? Create a separate constructor function? That's why at first I made this option negative so that it would always be enabled but I was told that you prefer positive options.

Reply:
Hm I have changed my mind, since now it requires disabling it I'm fine with it 👍 The user should be knowing what he is doing.